src/features/farming/hud/components/PlaceableController.tsx

```diff
@@ -21,7 +27,45 @@ export const PlaceableController: React.FC = () => {
   ] = useActor(child);
 
   const handleConfirmPlacement = () => {
+    // prevents multiple toasts while spam clicking place button
+    if (!child.state.matches("idle")) {
+      return;
+    }
+
     send("PLACE");
+
+    if (gameService.state.event.type === "building.constructed") {
```

Comment:
I would suggest omitting this code until we have a more elegant way to handle this.

The `PlaceableController` aims to be clueless of the consuming event. It feels like we are adding specific logic in here (i.e. buildings, chickens etc.) - this could lead to some unexpected behaviour between different features.

From a UX point of view, I feel we can show the crafting costs while placing the item - most games take this approach and show the crafting costs above the confirm/reject buttons

Reply:
I remove the toast logic for now.  I am open to suggestions if there are any better way to implement this.